x-pack/plugins/alerting/server/application/rule/transforms/transform_raw_actions_to_domain_actions.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { omit } from 'lodash';
+import { SavedObjectReference } from '@kbn/core/server';
+import { injectReferencesIntoActions } from '../../../rules_client/common';
+import { RuleAttributes } from '../../../data/rule/types';
+import { RawRule, RuleActionTypes } from '../../../types';
+import { RuleDomain } from '../types';
+
+interface Args {
+  ruleId: string;
+  actions: RuleAttributes['actions'] | RawRule['actions'];
+  isSystemAction: (connectorId: string) => boolean;
+  omitGeneratedValues?: boolean;
+  references?: SavedObjectReference[];
+}
+
+export const transformRawActionsToDomainActions = ({
```

Comment:
This function adds the `type` in all actions. The `type` is not persisted in ES.